nixos/modules/services/web-apps/pixivfe.nix

```diff
@@ -0,0 +1,124 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.pixivfe;
+in
+{
+  options.services.pixivfe = {
+    enable = lib.mkEnableOption "PixivFE, a privacy respecting frontend for Pixiv";
+
+    package = lib.mkPackageOption pkgs "pixivfe" { };
+
+    openFirewall = lib.mkEnableOption "open ports in the firewall needed for the daemon to function";
+
+    settings = lib.mkOption {
+      type = lib.types.nullOr (lib.types.attrsOf lib.types.anything);
```

Comment:
We should use rfc42 style settings here and define the ones that are used in the assertions.

Reply:
we want to use a submodule with the freeformType like eg https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/config/nix.nix#L180-L184 or https://github.com/NixOS/nixpkgs/commit/a37768ae9851b54f77e55501360dc6720dea6f87

Reply:
This is passed as environment variables, what format should I use?